### PR TITLE
OSSMDOC-702: Document gateway injection.

### PR DESCRIPTION
**Pull request by JStickler**
Version(s): 4.9 - 4.12

Issue: [OSSMDOC-702](https://issues.redhat.com/browse/OSSMDOC-702)

Link to docs preview:  http://file.bos.redhat.com/jstickle/OSSMDOC-702/service_mesh/v2x/ossm-traffic-manage.html#ossm-automatic-gateway-injection_traffic-management

QE review:
Peer review:

### CODE REVIEW

**Comment by prachiyadav** [comment]
Hi,

I have verified steps from document but did not get the expected output.

Steps i followed

- Installed SMCP 2.3
- Installed bookinfo
- followed document steps from "Managing traffic in your service mesh"


Attaching screen-shot of results.

Screenshot from 2022-10-15 00-35-30

**Comment by prachiyadav** [comment]
Attaching SMCP for your reference 

[servicemeshcontrolplane-basic.txt](https://github.com/openshift/openshift-docs/files/9789386/servicemeshcontrolplane-basic.txt)

File: modules/ossm-automatic-gateway-injection.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Using automatic injection for gateways lets you deploy and manage gateways independent from the `ServiceMeshControlPlane` resource. This lets you manage your gateways with your user applications. Using auto-injection for gateway deployments gives developers full control over the gateway deployment, while also simplifying operations. When a new upgrade is available, or a configuration has changed, you restart the gateway pods to update them. This makes the experience of operating a gateway deployment the same as operating sidecars.
+
+[NOTE]
+====
+Injection is disabled by default for the `ServiceMeshControlPlane` namespace, for example `istio-system`. As a security best practice, deploy gateways in a different namespace from the control plane.
+====
+
+.Prerequisites
+
+* The namespace must have the `istio-injection=enabled` label.
```
Comment (dgn) [inline modules/ossm-automatic-gateway-injection.adoc:20]: This is not required in OSSM

File: modules/ossm-automatic-gateway-injection.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* The namespace must have the injection webhook enabled.
```
Comment (dgn) [inline modules/ossm-automatic-gateway-injection.adoc:21]: Better: "the namespace must be a member of the mesh (as defined in the ServiceMeshMemberRoll or by creating a ServiceMeshMember resource)"

File: modules/ossm-automatic-gateway-injection.adoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+When deploying a gateway, you must opt-in to injection by adding an injection label or annotation to the gateway `deployment` object.
+
+.Example deployment with annotations
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
```
Comment (dgn) [inline modules/ossm-automatic-gateway-injection.adoc:29]: We should also list the `Service`, `Role` and `RoleBinding` here, as they're required

**Comment by dgn** [comment]
We'll need a full example for this to work, here's the one from the [upstream docs](https://istio.io/latest/docs/setup/additional-setup/gateway/):
```
apiVersion: v1
kind: Service
metadata:
  name: istio-ingressgateway
  namespace: istio-ingress
spec:
  type: LoadBalancer
  selector:
    istio: ingressgateway
  ports:
  - port: 80
    name: http
  - port: 443
    name: https
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: istio-ingressgateway
  namespace: istio-ingress
spec:
  selector:
    matchLabels:
      istio: ingressgateway
  template:
    metadata:
      annotations:
        # Select the gateway injection template (rather than the default sidecar template)
        inject.istio.io/templates: gateway
      labels:
        # Set a unique label for the gateway. This is required to ensure Gateways can select this workload
        istio: ingressgateway
        # Enable gateway injection. If connecting to a revisioned control plane, replace with "istio.io/rev: revision-name"
        sidecar.istio.io/inject: "true"
    spec:
      containers:
      - name: istio-proxy
        image: auto # The image will automatically update each time the pod starts.
---
# Set up roles to allow reading credentials for TLS
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: istio-ingressgateway-sds
  namespace: istio-ingress
rules:
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get", "watch", "list"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: istio-ingressgateway-sds
  namespace: istio-ingress
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: istio-ingressgateway-sds
subjects:
- kind: ServiceAccount
  name: default
```

**Comment by rh-tokeefe** [comment]
@prachiyadav and @dgn I've created a PR containing the content changes Daniel suggested. I will tag you on that PR for review.

File: modules/ossm-automatic-gateway-injection.adoc (hunk 1 above)
Comment (rh-tokeefe) [reply modules/ossm-automatic-gateway-injection.adoc:20]: update made

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.13` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.12. And any PR going into `main` must also target the latest version branch (enterprise-4.13).

If the update in your PR does NOT apply to version 4.13 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by rh-tokeefe** [comment]
@bergerhoffer After Julie left Red Hat, I copied this branch and continued working on the doc effort. The PR I worked on was previously merged into main: https://github.com/openshift/openshift-docs/pull/52191

I believe this PR can be closed. I think closing this was something I overlooked in the broader workflow. I'll try to keep that in mind should a similar situation arise in the future.

**Comment by bergerhoffer** [comment]
@rh-tokeefe Thanks for letting me know! I will go ahead and close this PR then.

**Pull request closed without merging**